コメント表示、作成

MyskYko opened the pull request:
## やったこと
### Viewについて
ほとんどがTweetの機構と同じで、コピペしました
コピペしただけではないものをまとめました

 - commentのidインクリメント処理
 - CommentCell...コメント数のラベルを消した
 - CommentViewController...TableViewをrxではなくextensionで設定している
 - MakeCommentViewController...TweetをCommentに変えただけ

### ViewModelについて
それぞれのViewControllerに応じたViewModelを作りました
コメントの対象となるツイートを遷移前前のViewControllerからもらっています

### その他
tweetの配列関係のところははtweet"s"の方が良いかなと思い変更しました


## 関連イシュー
#9 

## 注意点
このままだとコメント投稿後にでるCommentViewのTableが更新されていない
コメントの対象となるツイートをViewModelの引数で受け渡している

## レビューしてほしい点
 - [ ] CommentViewControllerがCommentViewModelのletを直接参照している点
 - [ ] ツイートをViewModelの引数で渡しているがこれで良いのか
 - [ ] コメントされるツイートがコメント作成時に削除されてしまった場合どうなるのか

この程度のプルリクコメントを書くのに30分ぐらいかかるのがウケます
 
![simulator screen shot 2017 04 29 14 50 03](https://cloud.githubusercontent.com/assets/18373300/25553507/df863474-2cef-11e7-9f62-9ea1c9463961.png)
![simulator screen shot 2017 04 29 14 50 07](https://cloud.githubusercontent.com/assets/18373300/25553508/e392f598-2cef-11e7-81e4-04f62bde2324.png)

MyskYko committed:
create view of commentView

MyskYko committed:
Merge branch 'master' into feature-comment

MyskYko committed:
ツイートを単体で表示する画面を作りました

MyskYko committed:
コメントの投稿ができるようになりました

MyskYko committed:
いらないファイルを消しました

yukitoto commented on Miyatter/Comment.swift:
```diff
@@ -21,4 +21,11 @@ class Comment: Object {
+        if let lastComment = realm.objects(Comment.self).sorted(byKeyPath: "id", ascending: true).last {
+            id = lastComment.id + 1
+        }
+    }
```
```
let comment = Comment()
comment.content = "" // クラッシュしない
realm.write {
   realm.add(comment)
}

comment.content = "" // クラッシュ
```

yukitoto replied:
```
if let realm = self.realm {
   if realm.inWriteTrasaction? {

   } else { 

   }

} else {
}

```

yukitoto commented on Miyatter/CommentCell.swift:
```diff
@@ -0,0 +1,75 @@
+        let label = UILabel()
+        label.font = UIFont(name: "HiraKakuProN-W3", size: 16)
+        label.numberOfLines = 0
+        label.lineBreakMode = .byCharWrapping
```
これ必要ない

yukitoto commented on Miyatter/CommentCell.swift:
```diff
@@ -0,0 +1,75 @@
+    }()
+    
+    
+    // MARK: - Life Cycle Events -
```
空行

yukitoto commented on Miyatter/CommentCell.swift:
```diff
@@ -0,0 +1,75 @@
+    // MARK: - Set Up Views -
+    
+    fileprivate func setUpView() {
+        self.addSubview(dateLabel)
```
selfがなくそう

yukitoto commented on Miyatter/CommentCell.swift:
```diff
@@ -0,0 +1,75 @@
+        dateLabel.snp.remakeConstraints { (make) in
+            make.top.equalTo(self).inset(8)
+            make.left.right.equalTo(self).inset(20)
+            make.height.equalTo(16)
```
labelのheight指定は極力しない

yukitoto commented on Miyatter/CommentViewController.swift:
```diff
@@ -0,0 +1,170 @@
+import RxSwift
+import RxCocoa
+
+class CommentViewController: UIViewController {
```
命名変更

yukitoto commented on Miyatter/CommentViewController.swift:
```diff
@@ -0,0 +1,170 @@
+            .tap
+            .subscribe(onNext: { [weak self] in
+                if let tweet = self?.viewModel.tweet {
+                    self?.present(MakeCommentViewController(viewModel: MakeCommentViewModel(selected: tweet)), animated: true, completion: nil)
```
短くしよう

yukitoto commented on Miyatter/CommentViewController.swift:
```diff
@@ -0,0 +1,170 @@
+}
+
+
+//tableView
```
// MARK: - TableViewDataSoure -

yukitoto commented on Miyatter/CommentViewModel.swift:
```diff
@@ -0,0 +1,29 @@
+import Foundation
+import RealmSwift
+import RxSwift
+
```
空きすぎ？

yukitoto commented on Miyatter/CommentViewModel.swift:
```diff
@@ -0,0 +1,29 @@
+    
+    // MARK: - Life Cycle Events -
+    
+    init(selected: Tweet) {
```
```
init(tweet: Tweet)

self.tweet = tweet
```

yukitoto replied:
```
init(tweetId: Int) {
  tweet = realm.object....
} 
```

yukitoto replied:
```
let tweetId: Int
var tweet: Tweet {
   realm.objectforprim...(key: tweetId)
}
```

yukitoto commented on Miyatter/MakeCommentViewController.swift:
```diff
@@ -0,0 +1,136 @@
+import UIKit
+import SnapKit
+import RxSwift
+import RxCocoa
```
RxCocoaはimport必要ない気がする

yukitoto commented on Miyatter/MakeCommentViewController.swift:
```diff
@@ -0,0 +1,136 @@
+import RxSwift
+import RxCocoa
+
+class MakeCommentViewController: UIViewController {
```
`Make` => `Create`

yukitoto commented on Miyatter/MakeCommentViewModel.swift:
```diff
@@ -0,0 +1,42 @@
+    // MARK: - Initializers -
+    
+    init(selected: Tweet) {
+        tweet = selected
```
ここも引数名を変更する

MyskYko committed:
・名前の変更をしました(CommentViewはTweetDetailView、MakeはCreate)
・setIdをwriteトランザクション内での呼び出すようにコメントを作成しました
・lineBreakを削除
・weakとunownedを交換
・Cellクラスのself.addSubViewのselfを削除
・ラベルのheightを削除
・コメント対象のTweetをIdで受け渡すようにしました
・tweetIdからインスタンスを作成するときオプショナルチェイニングします
・インデントやimportの修正

MyskYko commented:
## やったこと
思うんですが、これはコミットメッセージを見ていただければ良いのでは...

## 注意点
tweetをidで受け渡すようにしたのでインスタンス取得時にnilになる場合があり、ViewControllerの方でクラッシュしないように修正してあります

yukitoto commented on Miyatter/TweetDetailViewController.swift:
```diff
@@ -0,0 +1,177 @@
+}
+
+
+//MARK: - TableViewDataSource -
```
// のあとに空白かな

yukitoto commented on Miyatter/TimeLineViewController.swift:
```diff
@@ -103,15 +102,27 @@ class TimeLineViewController: UIViewController {
+        tableView.rx.itemSelected.subscribe(onNext: { [unowned self] IndexPath in
+            let tweet = self.viewModel.tweetsVariable.value[IndexPath.row]
+            let viewModel = TweetDetailViewModel(tweetId: tweet.id)
+            self.present(TweetDetailViewController(viewModel: viewModel),
```
present(のあとに改行入れたほうがきれいかも。

yukitoto commented on Miyatter/CreateCommentViewModel.swift:
```diff
@@ -0,0 +1,45 @@
+    // MARK: - Initializers -
+    
+    init(tweetId: Int) {
+        submitComment
```
init?(tweetId: Int)かな。
ViewController側でハンドリングするのが良い気がする。

MyskYko committed:
修正しました
・TweetDetailViewはinit?を使ってハンドリングするように変更しました
・インデント修正
